lib/db/mysql/index.js

```diff
@@ -388,19 +398,62 @@ MysqlStore.prototype = {
     });
   },
 
-  getAccessToken: function getAccessToken(tok) {
-    return this._readOne(QUERY_ACCESS_TOKEN_FIND, [buf(tok)]).then(function(t) {
+  /**
+   * Get an access token by token id
+   * @param id Token Id
+   * @returns {*}
+   */
+  getAccessToken: function getAccessToken(id) {
+    return this._readOne(QUERY_ACCESS_TOKEN_FIND, [buf(id)]).then(function(t) {
       if (t) {
         t.scope = t.scope.split(' ');
       }
       return t;
     });
   },
 
+  /**
+   * Remove token by token id
+   * @param id
+   * @returns {*}
+   */
   removeAccessToken: function removeAccessToken(id) {
     return this._write(QUERY_ACCESS_TOKEN_DELETE, [buf(id)]);
   },
 
+  /**
+   * Get all services that have have non-expired tokens
+   * @param {String} uid User ID as hex
+   * @returns {Promise}
+   */
+  getActiveClientTokensByUid: function getActiveClientTokensByUid(uid) {
+    if (! uid) {
+      return P.reject(new Error('Uid is required'));
+    }
+
+    return this._read(QUERY_ACTIVE_CLIENT_TOKENS_BY_UID, [
+      buf(uid)
+    ]);
+  },
+
+  /**
+   * Delete all non-expired tokens for some clientId and uid.
+   *
+   * @param {String} clientId Client ID
+   * @param {String} uid User Id as Hex
+   * @returns {Promise}
+   */
+  deleteActiveClientTokens: function deleteActiveClientTokens(clientId, uid) {
+    if (! clientId || ! uid) {
+      return P.reject(new Error('clientId and uid are required'));
```

Comment:
These error strings worried me (instead of using a proper error code, but then I remembered that none of the other methods do these kinds of checks. Are they required? I believe if the values are null, the query will search for those columns that are null. If the columns are `NOT NULL`, it should always return an result.